### PR TITLE
finished homework 17 (intro to spring framework)

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/main/java/com/spring/beans/Bicycle.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Setter
+@Getter
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
```
Comment (Taniann) [inline src/main/java/com/spring/beans/Bicycle.java]: singleton is a default scope for beans
so it is redundant annotation here

File: src/main/java/com/spring/beans/Bicycle.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+@PropertySource({"classpath:spring/bicycle.properties"})
+public class Bicycle implements Transport {
+    private static Logger log = Logger.getLogger(Car.class.getName());
```
Comment (Taniann) [inline src/main/java/com/spring/beans/Bicycle.java]: also add final modificator

File: src/main/java/com/spring/beans/Bicycle.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private int speed;
+
+    @Value("${countOfWheels}")
+    private int countOfWheels;
+
+    @Autowired
+    @Qualifier("summerWheel")
+    private Wheel wheel;
+
+    public Bicycle() {
```
Comment (Taniann) [inline src/main/java/com/spring/beans/Bicycle.java]: why dont you use lombok for its purpose?

File: src/main/java/com/spring/beans/Bicycle.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+public class Bicycle implements Transport {
+    private static Logger log = Logger.getLogger(Car.class.getName());
+
+    @Value("${speed}")
+    private int speed;
+
+    @Value("${countOfWheels}")
+    private int countOfWheels;
+
+    @Autowired
```
Comment (Taniann) [inline src/main/java/com/spring/beans/Bicycle.java:29]: for future read about DI with lombok 
https://www.baeldung.com/spring-injection-lombok

File: src/main/java/com/spring/beans/Car.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Setter
+@Getter
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@PropertySource({"classpath:spring/car.properties"})
+public class Car implements Transport {
+    private static Logger log = Logger.getLogger(Car.class.getName());
```
Comment (Taniann) [inline src/main/java/com/spring/beans/Car.java]: the same

File: src/main/java/com/spring/beans/SummerWheel.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package com.spring.beans;
+
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+@Component("summerWheel")
```
Comment (Taniann) [inline src/main/java/com/spring/beans/SummerWheel.java]: it is default value 
https://stackoverflow.com/questions/48871789/how-are-beans-named-by-default-when-created-with-annotation

**Review by Taniann — CHANGES_REQUESTED** [review]
pleasee resolve comments

File: src/main/java/com/spring/beans/SummerWheel.java (hunk 6 above)
Comment (Sofasmile) [reply src/main/java/com/spring/beans/SummerWheel.java]: It doesn`t work, when I delete this line.

File: src/main/java/com/spring/beans/Bicycle.java (hunk 4 above)
Comment (Sofasmile) [reply src/main/java/com/spring/beans/Bicycle.java:29]: Thank you!

File: src/main/java/com/spring/beans/SummerWheel.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.spring.beans;
+
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+@Component()
```
Comment (Taniann) [inline src/main/java/com/spring/beans/SummerWheel.java]: better without braces

**Review by Taniann — APPROVED** [review]
good

File: src/main/java/com/spring/beans/Bicycle.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    @Qualifier("summerWheel")
+    private Wheel wheel;
+
+    public Bicycle(Wheel wheel, int countOfWheels, int speed) {
```
Comment (Taniann) [inline src/main/java/com/spring/beans/Bicycle.java]: whu dont you want to use lombok annotation for its purpose?

Comment (Sofasmile) [reply src/main/java/com/spring/beans/Bicycle.java]: I resolved it

**Pull request merged**